### DISCUSSION TITLE
Welcome to "ML Foundations" Discussions!

### DISCUSSION THREAD

**Discussion by jonkrohn** [jonkrohn/ML-foundations#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mclr47** [comment]
2-linear-algebra-ii.ipynb

   .....very tiny observation , in the paragraph below should the slope
$m2$ be negative 2 instead of positive 2, since the values of y are decreasing while the values of x are increasing ?
~~~~~beginning of the paragraph~~~~~~~~
  Although it appears there is only one predictor ($x_1$), we need a second one (let's call it $x_0$) in order to allow for a $y$-intercept (therefore, $m$ = 2). Without this second variable, the line we fit to the plot would need to pass through the origin (0, 0). The $y$-intercept is constant across all the points so we can set it equal to 1 across the board:
      ~~~~~~end of the paragraph~~~~~~~~~~~

**Reply by jonkrohn** [reply]
Hi @mclr47 ! As per the slides/video, in this instance $m$ was intended to refer to the number of columns in the matrix $X$, which is equal to two because we have two variables in the model: $x_0$ (the $y$-intercept) and $x_1$ (the slope). 

However, I acknowledge this is confusing given that I re-use the variable $m$ as the slope later on in the example. As such, I have removed the confusing reference to $m$ as the number of columns.

**Comment by spiritmate** [comment]
Hi Jon! 

Thanks for the fantastic material, I'd like to ask that do we have the chance to download the presentation slides?

Thank you!

**Reply by jonkrohn** [reply]
Glad you're enjoying the material, @spiritmate! If you go to [jonkrohn.com/talks](https://www.jonkrohn.com/talks), you can access the particular slides you're looking for by finding recent "Machine Learning Foundations" classes I've taught. Lemme know if you can't find what you're looking for!

**Reply by spiritmate** [reply]
Hi @jonkrohn! 

Wow, that's perfect, thank you so much for the quick reply! :)

**Reply by ochuerta** [reply]
Hi, @jonkrohn . 

It looks like the page [jonkrohn.com/talks](https://github.com/jonkrohn/ML-foundations/discussions/jonkrohn.com/talks) no longer exist (getting a 404).  

Do you have another more recent link to these slides?  Thanks!

**Reply by jonkrohn** [reply]
Hi @ochuerta, the page exists! Try it again. There was an issue with the way GitHub was providing external links, but it is now resolved. Go here and you're all set: [https://www.jonkrohn.com/talks](https://www.jonkrohn.com/talks)

**Reply by ochuerta** [reply]
Excellent.  It works!  Thank you, @jonkrohn !

**Comment by suhermantom** [comment]
[ 6.statistics.ipynb - **Little Issue** ]
Hi @jonkrohn,

Great materials and course. I recommend everyone for Jon's book and online course at O'reilly.
The local library bought your Deep Learning Illustrated upon my request.
Anyway,
I follow along your courses on O'Reilly, and now I am learning '_**Probability and Statistics for Machine Learning**_'. I got little issue when running `beta = result.params` as I got different result, i.e. `array([3.5, 1.4])' when calling 'beta'. I have tried to rerun the above codes, but nothing change. What might be the issue? I cannot run the following code related to it. Please help.

**Reply by jonkrohn** [reply]
Hi @msuherma! I'm delighted your local library now has a copy of my book and that you're enjoying my videos on O'Reilly :)

Can you help me out with a bit more detail please? For example, what is the result that you were expecting? Are you referring perhaps to lines 230 or 265 in the ["Subject 6" Jupyter notebook](https://github.com/jonkrohn/ML-foundations/blob/master/notebooks/6-statistics.ipynb)? Are you using Google Colab to execute the notebook or do you have a notebook server running on your local machine?

**Comment by scottschmidl** [comment]
I came to this repository to check out some deeper/behind the scenes ML, but this repository moves sooooo slow through the videos. It's as if you are teach a Python beginner course. I had to stop watching. Recommendation: skip teaching beginner Python and move faster through the lessons. This is supposed to be a ML course not Python Fundamentals.

**Reply by jonkrohn** [reply]
Hi @scottschmidl! Thanks for the feedback, but I'm not sure I agree with your recommendation. 

As per the [repo README](https://github.com/jonkrohn/ML-foundations#machine-learning-foundations), particularly the ["ML House" section](https://github.com/jonkrohn/ML-foundations#machine-learning-foundations), this *ML Foundations* curriculum is not intended to cover ML directly. Rather, it is a beginner course on the foundational subjects that *underlie* ML, including at the onset a thorough introduction to tensor creation and manipulation in Python. At the very *end* of the curriculum, I leave learners with my recommended external resources for moving onward from these foundational subjects to ML proper. 

So perhaps this is not the right repo for you at all! Or perhaps you could simply skip past the content that you find slow or are already familiar with to dig into later topics which you might not be. My preference is to be methodical (read: slow) and thorough, and provide learners the opportunity to jump ahead if they desire. In contrast, if the learner *does* need what you happen to find slow and I don't provide it, they are left high and dry.

Keep on rockin'! I hope you find what you're looking for :)

**Reply by scottschmidl** [reply]
I appreciate your response. I really like the deep dive into the mathematics and algorithms, so I do think this is a useful resources for me.

**Reply by jonkrohn** [reply]
😀

**Comment by gphilpott-com** [comment]
Hi Jon, Thanks for putting this great course together! I’ve signed up to the Udemy course and downloaded the slides, all very helpful. I have a question regarding how to access the notebooks on GitHub exclusively via the web. I have used Google Colab and find it very easy to get going with, but I don’t love the editor. I am experimenting with GitHub Codespaces with Copilot and love the editor /copilot combo as it’s great for a “new” developer. (I’m running all this via the web on a iPad Pro) 

My question is, what is the best way to access your notebooks via codespaces so that I have a sandbox to play with your notebooks? I have cloned your repo to my private repo and then created a codespace from my private repo that contains your notebooks. Does that should about right? Is there a better way? I am new to all this and I do not want to inadvertently push a change to your repo 😬, just want a sandbox to learn in using your notebooks exercises.

Thanks,
Greg
